pyuvdata/uvdata/initializers.py

```diff
@@ -0,0 +1,564 @@
+            time_axis_faster_than_bls,
+        ) = utils.determine_rectangularity(
+            time_array=times, baseline_array=baselines, nbls=nbl, ntimes=nt
+        )
+
+    return (
+        nbl,
+        nt,
+        blts_are_rectangular,
+        time_axis_faster_than_bls,
+        times,
+        antpairs,
+        time_sized_arrays,
+    )
+
+
+def set_phase_params(obj, phase_center_catalog, phase_center_id_array, time_array):
+    """Configure phase center parameters for new UVData object."""
+    if phase_center_catalog is None:
+        obj._add_phase_center(cat_name="unprojected", cat_type="unprojected")
+    else:
+        for key, cat in phase_center_catalog.items():
+            print(f"Doing {key} {cat}")
+            obj._add_phase_center(cat_id=key, **cat)
+
+    if phase_center_id_array is None:
+        print("yes it is: ", len(obj.phase_center_catalog))
+        if len(obj.phase_center_catalog) > 1:
+            raise ValueError(
+                "If phase_center_catalog has more than one key, phase_center_id_array "
+                "must be provided."
+            )
+
+        phase_center_id_array = np.full(
+            len(time_array), next(iter(obj.phase_center_catalog.keys())), dtype=int
+        )
+    obj.phase_center_id_array = phase_center_id_array
+    obj._set_app_coords_helper()
+
+
+def get_spw_params(
+    flex_spw_id_array: np.ndarray | None = None, freq_array: np.ndarray | None = None
+) -> tuple[np.ndarray, np.ndarray]:
+    """Configure spectral window parameters for new UVData object."""
+    if flex_spw_id_array is None:
+        flex_spw_id_array = np.zeros(freq_array.shape[0], dtype=int)
+        spw_array = np.array([0])
+    else:
+        spw_array = np.sort(np.unique(flex_spw_id_array))
+
+    return flex_spw_id_array, spw_array
+
+
+def new_uvdata(
+    freq_array: np.ndarray,
+    polarization_array: np.ndarray | list[str | int] | tuple[str | int],
+    antenna_positions: np.ndarray | dict[str | int, np.ndarray],
+    telescope_location: Locations,
+    telescope_name: str,
+    times: np.ndarray,
+    antpairs: Sequence[tuple[int, int]] | np.ndarray | None = None,
+    do_blt_outer: bool | None = None,
+    integration_time: float | np.ndarray | None = None,
+    channel_width: float | np.ndarray | None = None,
+    antenna_names: list[str] | None = None,
+    antenna_numbers: list[int] | None = None,
+    blts_are_rectangular: bool | None = None,
+    data_array: np.ndarray | None = None,
+    flag_array: np.ndarray | None = None,
+    nsample_array: np.ndarray | None = None,
+    flex_spw_id_array: np.ndarray | None = None,
+    history: str = "",
+    instrument: str = "",
+    vis_units: Literal["Jy", "K str", "uncalib"] = "uncalib",
+    antname_format: str = "{0:03d}",
+    empty: bool = False,
+    time_axis_faster_than_bls: bool | None = None,
+    phase_center_catalog: dict[str, Any] | None = None,
+    phase_center_id_array: np.ndarray | None = None,
```

Comment:
A slightly more user-friendly approach here would be to also accept the same inputs as the `phase` method -- a phase_center_catalog can be constructed from those inputs as is done in that method. Or we can punt on that until a user asks for it.

Reply:
I think I'm gonna punt on that, because I really do see the most user-friendly approach as being to provide a phase_center_catalog *object*, and I don't want to then have to deprecate all those `phase` params.